src/dmd/expressionsem.d

```diff
@@ -12807,6 +12818,31 @@ bool checkAddressVar(Scope* sc, Expression exp, VarDeclaration v)
     return true;
 }
 
+/****************************************************
+ * Determine if the address of a `ref return` value of
+ * a function call with type `tf` can be taken safely.
+ *
+ * This is currently stricter than necessary: it can be safe to take the
+ * address of a `ref` with pointer type when the pointer isn't `scope`, but
+ * that involves inspecting the function arguments and parameter types which
+ * is more complex.
```

Comment:
This kind of inspection is just what escape.d is crafted to do. I'd prefer this be done.

Reply:
In that case I could use some help, because I don't know how to ask escape.d if a `CallExp` returns scope pointers.